autogluon/utils/tabular/utils/loaders/load_pkl.py

```diff
@@ -22,8 +27,19 @@ def load(path, format=None, verbose=True):
         return pickle.loads(s3.Bucket(s3_bucket).Object(s3_prefix).get()['Body'].read())
 
     if verbose: logger.log(15, 'Loading: %s' % path)
-    with open(path, 'rb') as fin:
-        object = pickle.load(fin)
+
+    compression_fn_map = compression_utils.get_compression_map()
+    validated_path = compression_utils.get_validated_path(path, compression_fn=compression_fn)
+
+    if compression_fn_kwargs is None:
+        compression_fn_kwargs = {}
+
+    if compression_fn in compression_fn_map:
+        with compression_fn_map[compression_fn]['open'](validated_path, 'rb', **compression_fn_kwargs) as fin:
+            object = pickle.load(fin)
+    else:
+        raise ValueError(f'compression_fn={compression_fn} or compression_fn_kwargs={compression_fn_kwargs} are not valid. Valid function values: {compression_fn_map.keys()}')
```

Comment:
We are still not validating `compression_fn_kwargs` here, do you think it is worth doing?  I decided not to since a change to the various compression function open API's could break things.

Reply:
It's fine to not validate kwargs, as that is hard to do.